eng/scripts/New-BuildInfo.ps1

```diff
@@ -36,6 +33,20 @@ $nativePlatforms = @(
 #     $nativePlatforms = @('linux-x64-native')
 # }
 
+if ($BuildId -eq 0) {
+    if ($isPipelineRun) {
+        LogError 'A non-zero BuildId is required when running in a pipeline.'
+        $exitCode = 1
+    } else {
+        $BuildId = 99999
+    }
+}
+
+if ($isPipelineRun -and !$PublishTarget) {
+    LogError 'PublishTarget parameter is required when running in a pipeline.'
+    $exitCode = 1
```

Comment:
Should these be direct exit's? It looks like the exitcode could get overwritten otherwise. Although you are only setting it 1 below.